src/main/java/util/Parser.java

```diff
@@ -0,0 +1,171 @@
+                    return Command.DELETE;
+                } catch (NumberFormatException e) {
+                    //System.out.println("Caught nfe");
+                    throw new DukeException("Sorry, I don't think that's a valid number...");
+                }
+            } else {
+                // Something wrong with command, throw exception
+                throw new DukeException("Whoops! Something went wrong and I can't process your "
+                        + instructions[0] + " command. Sorry! D:");
+            }
+        } else if (instructions[0].equals("find")) {
+            // Command: "find <keyword>"
+            if (hasCmdDetails(instructions)) {
+                return Command.FIND;
+            } else {
+                // Something wrong with command, throw exception
+                throw new DukeException("Whoops! Something went wrong and I can't process your "
+                        + instructions[0] + " command. Sorry! D:");
+            }
+        }
+        return Command.INVALID;
+    }
+
+    /**
+     * Returns lateral location of the specified position.
+     * If the position is unset, NaN is returned.
+     *
+     * @param cmd String array with the command sectioned out.
+     * @return True/False value of whether the command is complete.
+     * @throws DukeException  If command has missing data.
+     */
+    public static boolean hasCmdDetails(String[] cmd) throws DukeException {
+        try {
+            if (cmd[1].equals("") || cmd[1].trim().length() == 0) {
+                /** Make sure command has follow up details **/
+                // Check if there is second word
+                // Check if second word is just whitespace
+                // If so, command has no details
+                return false;
+            }
+            return true;
+        } catch (ArrayIndexOutOfBoundsException e) {
+            // Check if there is even anything after command
+            // Exception thrown if userInput.split() fails
+            // Command has no details, throw exception
+            throw new DukeException("Whoops! I think you forgot to finish your "
+                    + "command. Sorry but I need it. D:");
+        }
+    }
+
+    /**
+     * Returns lateral location of the specified position.
+     * If the position is unset, NaN is returned.
+     *
+     * @param dateString String that may be a LocalDate.
+     * @throws DukeException  If dateString is not a LocalDate.
+     */
+    public static void checkDate(String dateString) throws DukeException {
+        try {
+            LocalDate df = LocalDate.parse(dateString);
+        } catch (Exception e) {
+            // Check if string can be recognized as a valid LocalDate
+            // If can't, print out error message
+            throw new DukeException("Whoops! I think there is an error in your date." +
+                    "\nPlease Try Again!");
+        }
+    }
+}
```

Comment:
Seems like the purpose of this function is to catch an exception and throw a different exception in its place. Not sure if this is good practice as it may mask errors.